include/deal.II/matrix_free/fe_evaluation_data.h

```diff
@@ -430,16 +447,29 @@ class FEEvaluationData
   get_subface_index() const;
 
   /**
-   * If is_face is true, this function returns the orientation index within an
-   * array of orientations as stored in ShapeInfo for unknowns and quadrature
-   * points.
+   * If is_face is true, this function returns for a given lane the
+   * orientation index within an array of orientations as stored in
+   * ShapeInfo for unknowns and quadrature points.
    *
-   * @note This function depends on the internal representation of data, which
-   * is not stable between releases of deal.II, and is hence mostly for
-   * internal use.
+   * @note Only available for `dof_access_index == dof_access_cell` and
+   * `is_interior_face == false`.
    */
-  unsigned int
-  get_face_orientation() const;
+  std::uint8_t
+  get_face_orientation(const unsigned int v = 0) const
+  {
+    // implemented inline to avoid compilation problems on Windows
+    Assert(is_face, ExcNotInitialized());
+    Assert(v == 0 ||
+             (cell != numbers::invalid_unsigned_int &&
+              dof_access_index ==
+                internal::MatrixFreeFunctions::DoFInfo::dof_access_cell &&
+              is_interior_face == false),
+           ExcMessage(
+             "All face numbers can only be queried for ECL at exterior "
+             "faces. Use get_face_no() in other cases."));
+
+    return face_orientations[v];
+  }
```

Comment:
May I suggest to move the implementation of this function and the one above to outside the class declaration (down where the `get_face_orientation()` was originally implemented)? The inline definition was only used because I could not easily return an array and be compatible with `Number == double`, as there is not always a `Number::size()` then; in addition, the MSVC compiler did not like accessing the class definition of `n_lanes`, so I stopped trying. So now we can reduce this workaround to two very simple functions.